analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -187,13 +230,14 @@ public override void ExecutionCompleted()
     private ProgramState ProcessInvocation(SymbolicContext context, IInvocationOperationWrapper invocation)
     {
         var targetMethod = invocation.TargetMethod;
-        if (targetMethod.Is(KnownType.System_Linq_Enumerable, nameof(Enumerable.Count))
-            && SizeConstraint(context.State, invocation.Instance ?? invocation.Arguments[0].ToArgument().Value, HasFilteringPredicate()) is { } constraint)
-        {
-            return context.SetOperationConstraint(constraint);
-        }
-        else if (invocation.Instance is { } instance)
```

Comment:
After removing this, `invocaton.Instance` can be `null`. And it was used inside the original body of the if.

Reply:
Wait, that check is still there:
`if ((invocation.Instance ?? invocation.Arguments.FirstOrDefault()?.ToArgument().Value) is { } instance)`